### PR TITLE
Generate targets files that throws for unsupported netstandard applicable tfms

### PR DESCRIPTION
**Pull request by ViktorHofer**
For S.D.Common, S.Speech and S.Sec.Crypto.Pkcs we manually added a
targets file to mark the .NETStandard asset as not applicable.
This was done to allow defining a minimum supported .NETCoreApp
version, even though a compatible .NETStandard asset is available.
This commit automatically generates that targets file based on
items.

This is a port of the commit from https://github.com/dotnet/runtime/pull/53023.

### CODE REVIEW

**Review by Anipik — APPROVED** [review]

File: Directory.Build.targets [hunk 1]
```diff
@@ -70,4 +71,39 @@
...
+                      Encoding="Unicode" />
+
+    <ItemGroup>
+      <None Include="$(_NETStandardCompatErrorFilePath)"
+            PackagePath="buildTransitive\%(NETStandardCompatError.Identity)"
+            Pack="true" />
+      <None Include="$(PlaceholderFile)"
+            PackagePath="buildTransitive\%(NETStandardCompatError.Supported)"
+            Pack="true" />
+      <FileWrites Include="$(_NETStandardCompatErrorFilePath)" />
```
Comment (ericstj) [inline Directory.Build.targets:106]: Clean may not work correctly for pkgproj, but test this out for csproj if you have a chance.  Clean does some weird things with paths based on where they are wrt output path

**Review by ericstj — APPROVED** [review]

File: Directory.Build.targets [hunk 2]
```diff
@@ -70,4 +71,39 @@
...
+    <Error Text="$(PackageId) doesn't support %24(TargetFramework). Consider updating your TargetFramework to %(NETStandardCompatError.Supported) or later." />
+  </Target>
+</Project>]]>
+      </_NETStandardCompatErrorFileContent>
+    </PropertyGroup>
+
+    <WriteLinesToFile File="$(_NETStandardCompatErrorFilePath)"
+                      Lines="$(_NETStandardCompatErrorFileContent)"
+                      WriteOnlyWhenDifferent="true"
+                      Encoding="Unicode" />
```
[the remark this answers is not in the document]
Comment (ViktorHofer) [reply Directory.Build.targets:97]: Nope, just copied the code frkm the package validation component in the SDK... 😁

**Comment by ViktorHofer** [comment]
Merging as the allconfigurations builds passed. wasm failures are likely: https://github.com/dotnet/runtime/pull/53180

File: Directory.Build.targets (hunk 2 above)
[the remark this answers is not in the document]
Comment (ericstj) [reply Directory.Build.targets]: Interesting, I wonder if they had a reason.  Will check the history.

[the remark this answers is not in the document]
Comment (ViktorHofer) [reply Directory.Build.targets]: Sorry I was unclear. I meant the one that @Anipik added to the PackageValidation.targets file. @Anipik was there a specific reason to use Unicode Encoding?

[the remark this answers is not in the document]
Comment (ericstj) [reply Directory.Build.targets]: That was clear.  I went and read through history and the PR and didn't see any discussion around it.  Also looked at other uses of WriteLinesToFile in the SDK and didn't see them using Unicode.  Perhaps it was arbitrary?  Would be interesting in @Anipik's thoughts.

**Pull request merged**